src/test/tcp/test_tcp.c

```diff
@@ -49,6 +61,37 @@ static void _mylog(const char* fileName, const int lineNum, const char* funcName
     fflush(stdout);
 }
 
+// create a System V queue with key based on a file queue name
+static int get_queue(const char* queuename) {
+    key_t key = ftok(queuename, 0);
+    assert_true_errno(key != -1);
+
+    int fd = msgget(key, IPC_CREAT | 0666);
+    assert_true_errno(fd != -1);
+
+    return fd;
+}
+
+static void queue_send_u16(const char* queuename, uint16_t i) {
+    int queue, r;
+    IntegerMessage msg = {QUEUE_MTYPE, i};
+
+    queue = get_queue(queuename);
+    r = msgsnd(queue, &msg, sizeof(msg.msg), IPC_NOWAIT);
```

Comment:
We don't want `IPC_NOWAIT` here. Better to block. We have timeouts for the test as a whole that will prevent the test from blocking forever if something goes wrong.